src/faivor/metrics/metric.py

```diff
@@ -0,0 +1,24 @@
+from dataclasses import dataclass, field
+from typing import Callable, Dict
+from abc import ABC
+import torch
+
+@dataclass
+class ModelMetric(ABC):
```

Comment:
```suggestion
import torch

@dataclass
class ModelMetric():
```
Is there a particular reason to make the class abstract? We are instantiating it later in the code, so I assume that it was not attended to be an abstract class.